zipline/testing/fixtures.py

```diff
@@ -292,3 +299,126 @@ def init_class_fixtures(cls):
         start_loc = end_loc - cls.TRADING_DAY_COUNT
 
         cls.trading_days = all_days[start_loc:end_loc + 1]
+
+
+class WithPipelineEventDataLoader(WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing common test methods/behaviors for event
+    data loaders.
+
+    `get_sids` must return the sids being tested.
+    `get_dataset` must return {sid -> pd.DataFrame}
+    `loader_type` must return the loader class to use for loading the dataset
+    `make_asset_finder` returns a default asset finder which can be overridden.
+    """
+    @abc.abstractproperty
```

Comment:
these should probably be class methods with reasonable defaults which can be overridden by the user of the fixture.